pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -13,6 +13,65 @@ import (
 	"github.com/holiman/uint256"
 )
 
+
+// getHighLen calculates the highest bit length of `scalar_u.d2` and `scalar_v.d2`,
+// subtracts 1 from the result, and assigns it to `ids.len_hi`. 
+//
+// `newGetHighLenHint` takes three operanders as arguments:
+//   - `len_hi`: the variable that will store the result of the bit-length calculation.
+//   - `scalar_u.d2`: the first scalar value.
+//   - `scalar_v.d2`: the second scalar value.
+//
+// The function resolves these operanders, calculates the maximum bit length of `scalar_u.d2` and `scalar_v.d2`,
+// and assigns the result (bit_length - 1) to `len_hi`.
+func newGetHighLenHint(len_hi, scalar_u, scalar_v hinter.ResOperander) hinter.Hinter {
+    return &hinter.GenericZeroHinter{
+        Name: "GetHighLen",
+        Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+            //> ids.len_hi = max(ids.scalar_u.d2.bit_length(), ids.scalar_v.d2.bit_length())-1
+
+            scalarU, err := hinter.ResolveAsFelt(vm, scalar_u)
+            if err != nil {
+                return fmt.Errorf("failed to resolve scalar_u.d2: %w", err)
+            }
+
+            scalarV, err := hinter.ResolveAsFelt(vm, scalar_v)
+            if err != nil {
+                return fmt.Errorf("failed to resolve scalar_v.d2: %w", err)
+            }
+
+            bitLenU := scalarU.BitLen()
+            bitLenV := scalarV.BitLen()
+            lenHi := utils.Max(bitLenU, bitLenV) - 1
+
+            lenHiAddr, err := len_hi.GetAddress(vm)
+            if err != nil {
+                return fmt.Errorf("failed to get address of len_hi: %w", err)
+            }
+
+            lenHiMv := memory.MemoryValueFromInt(lenHi)
+
+            return vm.Memory.WriteToAddress(&lenHiAddr, &lenHiMv)
+        },
+    }
+}
+
+func createGetHighLenHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
+    len_hi, err := resolver.GetResOperander("len_hi")
+    if err != nil {
+        return nil, err
+    }
+    scalar_u, err := resolver.GetResOperander("scalar_u.d2")
```

Comment:
Regarding previous comments, you just resolve as `scalar_u` and `scalar_v`, which you later resolve as `BigInt3`.